core/src/text/character_style.rs

```diff
@@ -0,0 +1,41 @@
+use crate::pixelcolor::PixelColor;
+
+/// Character style.
+///
+/// This trait is used to modify character styles programmatically, for example, to implement
+/// rendering of text with multiple colors. Applications shouldn't use this trait and instead use
+/// the character style types that are provided by the text renderer, like `MonoTextStyle` and
+/// `MonoTextStyleBuilder` for the integrated font support.
+///
+///  # Implementation notes
+///
+/// Text renderers don't need to support all settings in this trait. All calls to unsupported
+/// setters should be ignored by the implementation. The trait provided empty default
+/// implementations for all setters.
+pub trait CharacterStyle {
```

Comment:
@rfuest is there any chance we could require `CharacterStyle: Clone` (or even Copy)?

because e-t can change the char style, and `draw` takes a shared reference, I think e-t should be able to draw the same thing when `draw` is called multiple times. Because of this, I need to clone the character style (well, the whole TextRenderer in fact). This requires me to add a `Clone` bound so there's a chance certain font renderers won't be compatible with e-t.

Reply:
@rfuest friendly ping. Nothing urgent, but maybe this fell off the radar.

Reply:
Sorry, I hadn't noticed this.

The main disadvantage of adding `Clone` as a supertrait to a trait is that it would be no longer object safe. But because `CharacterStyle` has an associated type it isn't object safe anyway. So I think there should be no harm in adding `Clone`. I would choose `Clone` over `Copy`, because it's more universal and `style.clone()` should do the same as `*style` if `Copy` is also implemented.